### ISSUE TITLE
Issue with LT14 and BigQuery PipeSyntax

### ISSUE THREAD

**Issue by antoine-lefebvre** [sqlfluff/sqlfluff#6984]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Now that BigQuery Pipe syntax is supported, we have started to use it in my company. 
I found an issue recently regarding the WHERE clauses using a pipe

Here is an example of valid statement:
```SQL
FROM my_database.my_table
|> WHERE some_condition 
```

From a layout point of view this is perfectly fine. However according to LT14 rule, there should be a new line before the keyword
> L:   9 | P:   4 | LT14 | The 'WHERE' keyword should always start a new line. 

Running `sqlfluff fix` produces this output:
```SQL
FROM my_database.my_table
|> 
WHERE some_condition 
```
Which is not particularly pretty.  

Would it be possible not activate LT14 if the keyword is preceeded by a pipe operator?
Additionally, LT14 is not triggered when a pipe operator is not preceeded by a new line. i.e. this below is currently considered valid:
```SQL
FROM my_database.my_table |> 
WHERE some_condition 
```

### Expected Behaviour

```SQL
FROM my_database.my_table
|> WHERE some_condition 
```
Should not trigger LT14.

### Observed Behaviour

```SQL
FROM my_database.my_table
|> WHERE some_condition 
```
Triggers LT14 and sqlfluff fix produces:
```SQL
FROM my_database.my_table
|> 
WHERE some_condition 
```

### How to reproduce

run sqlfluff fix/lint on the following query:
```SQL
FROM my_database.my_table
|> WHERE some_condition 
```

### Dialect

bigquery

### Version

sqlfluff, version 3.4.1

### Configuration

default

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by antoine-lefebvre** [comment]
@WittierDinosaur  : maybe you could help here? 🙏🏻

**Comment by keraion** [comment]
Going to change this from a bug to enhancement as LT14 no longer splits the `pipe_operator` and `keywords`
